manticore/platforms/linux.py

```diff
@@ -2648,6 +2649,9 @@ def sys_unlink(self, pathname) -> int:
             return -e.errno
         return 0
 
+    # A cache for keeping state when reading directories { fd: dent_iter }
+    sys_getdents_c = {}
```

Comment:
I would love suggestions on the best way to do this. I want this to be a static variable for the `getdents` system call because it needs to store state. I saw that decorators could accomplish this, but do we do anything like this in Manticore that I could look at for inspiration?

Reply:
Are you sure you _really_ want there to be just one copy of this dict for _all_ the states? I'm not sure exactly what you're trying to accomplish here, but if you're just trying to store data in the `Linux` object, you can set `self._getdents_c = {}` in the `__init__` function as normal, then add corresponding entries to the `__getstate__` and `__setstate__` methods so that it's properly handled during state transitions.

Reply:
I do not want one copy of the dict for _all_ states. I'd like to do as you described, and I tried it a couple times, although I guess I missed adding the field to both `__getstate__` and `__setstate__`. It seems to work now. Thank you!